examples/bash_automatic_cd.sh

```diff
@@ -7,14 +7,14 @@
 # To undo the effect of this function, you can type "cd -" to return to the
 # original directory.
 
-function ranger-cd {
-    tempfile="$(mktemp -t tmp.XXXXXX)"
-    ranger --choosedir="$tempfile" "${@:-$(pwd)}"
-    test -f "$tempfile" &&
-    if [ "$(cat -- "$tempfile")" != "$(echo -n `pwd`)" ]; then
-        cd -- "$(cat "$tempfile")"
+ranger-cd() {
+    local temp_file chosen_dir
+    temp_file="$(mktemp)"
+    ranger --choosedir="$temp_file" -- "${@:-$PWD}"
+    if chosen_dir="$(<"$temp_file")" && [[ -n "$chosen_dir" ]]; then
+        cd -- "$chosen_dir"
```

Comment:
POSIX says nothing about `--` for `cd` so let's drop it.

Reply:
1. This was here before I edited this file.
2. Yes, [specification](https://pubs.opengroup.org/onlinepubs/009695399/utilities/cd.html) doesn't specify `--`, although this can lead to bugs when a directory is named `-L` or `-P`. Is this small bug negligible for POSIX-compliance?

Reply:
Hmm, I'm not sure, let's lean towards the status quo then.

Reply:
https://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap12.html#tag_12_02

> <dt><b>Guideline&nbsp;10:</b></dt>
> <dd>The argument <b>--</b> should be accepted as a delimiter indicating the end of options. Any following arguments should be treated as operands, even if they begin with the <tt>'-'</tt> character. The <b>--</b> argument should not be used as an option or as an operand.</dd>

As far as I can tell, support for `--` is specified in this document.

Reply:
Good enough for me.